modules/src/clients/ics07_tendermint/client_def.rs

```diff
@@ -193,116 +197,198 @@ impl ClientDef for TendermintClient {
 
     fn verify_client_consensus_state(
         &self,
-        _client_state: &Self::ClientState,
-        _height: Height,
-        _prefix: &CommitmentPrefix,
-        _proof: &CommitmentProofBytes,
-        _client_id: &ClientId,
-        _consensus_height: Height,
-        _expected_consensus_state: &AnyConsensusState,
+        client_state: &Self::ClientState,
+        prefix: &CommitmentPrefix,
+        proof: &CommitmentProofBytes,
+        root: &CommitmentRoot,
+        client_id: &ClientId,
+        consensus_height: Height,
+        expected_consensus_state: &AnyConsensusState,
     ) -> Result<(), Ics02Error> {
-        todo!()
+        let path = Path::ClientConsensusState {
```

Comment:
I think we're missing the following checks for all `verify_*()` methods ->
```
// check that the client is at a sufficient height
assert(clientState.latestHeight >= height)
// check that the client is unfrozen or frozen at a higher height
assert(clientState.frozenHeight === null || clientState.frozenHeight > height)
```
The first part of the second assertion (i.e. `clientState.frozenHeight === null`) is already checked, so we only need to check the second part -> https://github.com/informalsystems/ibc-rs/blob/6d54a5ac5597a3b08d2384c3849204c8fbf0de7f/modules/src/core/ics03_connection/handler/verify.rs#L133-L143

Also wondering what the best place for this check would be - should we have a separate function that all `verify_*()` functions call?

Reply:
Thank you!
I think that they should be called in `ics02_client::client_def::ClientDef` because the height validations are also required for other clients.
I'd like to add frozen_height() function at #1619. Then, I will add the height validations.

Reply:
Thanks! Sounds like a plan! :) I just reviewed #1619 and added some comments.